code/src/Consumable.ts

```diff
@@ -0,0 +1,38 @@
+import { Item } from './Item';
+
+export abstract class Consumable extends Item {
+  consumed: boolean = false;
+  spoiled: boolean;
+
+  constructor(name: string, value: number, weight: number, spoiled: boolean) {
+    super(name, value, weight);
+    this.spoiled = spoiled;
+  }
+
+  eat(): string {
```

Comment:
From my understanding, this method should only return "you eat..." and the main implementation should be in "use" method.
By the way, you don't need to implement "eat" in Consumable class because it is abstract and other classes such as Pizza will override it.